zad4/app/services/football_player_manager.rb

```diff
@@ -8,14 +8,20 @@ def initialize(player)
   end
 
   def stronger_players
-    # TODO: Implement me pls
+    Player.where('power > ?', player.power)
   end
 
   def set_power(value)
-    # TODO: I need body
+    Player.update(player.id, power: value)
   end
 
   def copy_player
-    # TODO: I feel empty inside
+    player_cp = @player.clone # or should we use dup?
```

Comment:
Lepszy w tym wypadku byłby `dup` ;) clone jest turbo dokładną kopią obiektu, wskazuje nawet na ten sam rekord w bazie, w wyniku czego save wywołany na nim nie tworzy nowego rekordu, a jedynie robi update na istniejącym obiekcie. `dup` natomiast generuje nowy obiekt, który jest płytką kopią. Głównie kopiuje same atrybuty i najważniejsze jest w nim to że nie jest związany z żadnym rekordem w bazie, w wyniku czego `save!` na takim obiekcie stworyz nowy rekord w bazie :)

Reply:
Jak w konsoli odpalisz komendę `rspec` to testy Ci też powiedzą że coś jest nie tak z tym save :D